lib/iris/experimental/regrid.py

```diff
@@ -1312,3 +1313,401 @@ def regridder(self, src_grid, target_grid):
+
+        for index in np.ndindex(tuple(iter_shape)):
+            src_index = list(index)
+            src_index[xy_dim] = slice(None)
+            src_subset = src_data[tuple(src_index)]
+            tgt_index = index[:xy_dim] + (slice(None), slice(None)) \
+                + index[xy_dim+1:]
+            data[tgt_index] = scipy.interpolate.griddata(
+                projected_src_points[..., :2], src_subset,
+                (projected_tgt_grid[..., 0], projected_tgt_grid[..., 1]),
+                method=method)
+        data = np.ma.array(data, mask=np.isnan(data))
+        return data
+
+    def _create_cube(self, data, src, src_xy_dim, src_x_coord, src_y_coord,
+                     grid_x_coord, grid_y_coord,
+                     regrid_callback):
+        """
+        Return a new Cube for the result of regridding the source Cube onto
+        the new grid.
+
+        All the metadata and coordinates of the result Cube are copied from
+        the source Cube, with two exceptions:
+            - Grid dimension coordinates are copied from the grid Cube.
+            - Auxiliary coordinates which span the grid dimensions are
+              ignored, except where they provide a reference surface for an
+              :class:`iris.aux_factory.AuxCoordFactory`.
+
+        Args:
+
+        * data:
+            The regridded data as an N-dimensional NumPy array.
+        * src:
+            The source Cube.
+        * src_xy_dim:
+            The X and Y dimension within the source Cube.
+        * src_x_coord:
+            The X :class:`iris.coords.DimCoord`.
+        * src_y_coord:
+            The Y :class:`iris.coords.DimCoord`.
+        * grid_x_coord:
+            The :class:`iris.coords.DimCoord` for the new grid's X
+            coordinate.
+        * grid_y_coord:
+            The :class:`iris.coords.DimCoord` for the new grid's Y
+            coordinate.
+        * regrid_callback:
+            The routine that will be used to calculate the interpolated
+            values of any reference surfaces.
+
+        Returns:
+            The new, regridded Cube.
+
+        """
+        # Create a result cube with the appropriate metadata
+        result = iris.cube.Cube(data)
+        result.metadata = copy.deepcopy(src.metadata)
+
+        # Copy across all the coordinates which don't span the grid.
+        # Record a mapping from old coordinate IDs to new coordinates,
+        # for subsequent use in creating updated aux_factories.
+        coord_mapping = {}
+
+        def copy_coords(src_coords, add_method):
+            for coord in src_coords:
+                dims = src.coord_dims(coord)
+                if coord is src_x_coord:
+                    coord = grid_x_coord
+                    # Increase dimensionality to account for 1D coord being
+                    # regridded onto 2D grid
+                    dims = list(dims)
+                    dims[0] += 1
+                    dims = tuple(dims)
```

Comment:
Since we know `dims` is of length 1, this can be reduced to `dims = (dims[0] + 1,)`, although maybe it's worth being verbose for clarity